core/modules/entity/entity.module

```diff
@@ -25,6 +25,41 @@ function entity_modules_uninstalled($modules) {
   // It is important that the entity info cache gets flushed last to prevent
   // fatal errors with outdated entity information.
   backdrop_register_shutdown_function('entity_info_cache_clear');
+
+  // Clean up any leftover field configuration files for entities defined by the
+  // uninstalled modules.
+  $field_names = array();
+  foreach ($modules as $module) {
+    $hook = $module . '_entity_info';
+    if (function_exists($hook)) {
+      $entity_infos = $hook();
+      foreach ($entity_infos as $entity_type => $entity_info) {
+        if (!empty($entity_info['fieldable'])) {
+          // Get any leftover instance config files.
+          $instance_config_names = config_get_names_with_prefix('field.instance.' . $entity_type . '.');
+          foreach ($instance_config_names as $instance_config_name) {
+            // Remove the instance config.
+            $config = config($instance_config_name);
```

Comment:
This doesn't care about field data, right?

So whatever's in the database at this point will stay there - without any way to purge, as with the next line the instance config is gone. And with it the info about entity type and bundle.

As long as we have the instance conf, we'd know the entity type and bundle name - regardless of the actual entity definition (which isn't available anymore).

Reply:
> So whatever's in the database at this point will stay there - without any way to purge, as with the next line the instance config is gone. And with it the info about entity type and bundle.

As mentioned in the issue, the database tables are all removed normally. All of this processing here is happening afterward. There are no tables remaining.

Didn't try the deleted property, but I assumed that, since the module is disabled and uninstalled, the delete will be ignored.

Reply:
Right, can be marked as resolved :+1:  (GH perms don't allow me that).